lib/finch/http1/pool.ex

```diff
@@ -280,16 +386,22 @@ defmodule Finch.HTTP1.Pool do
   def handle_cancelled(:queued, _pool_state), do: :ok
 
   defp transfer_if_open(conn, state, {pid, _} = from) do
+    transfer_if_open(conn, state, from, pid)
+  end
+
+  defp transfer_if_open(conn, state, from, pid) do
     if Conn.open?(conn) do
-      if state == :fresh do
-        NimblePool.update(from, conn)
+      case state do
+        :fresh ->
+          NimblePool.update(from, conn)
 
-        case Conn.transfer(conn, pid) do
-          {:ok, conn} -> {:ok, conn}
-          {:error, _, _} -> :closed
-        end
-      else
-        {:ok, conn}
+          case Conn.transfer(conn, pid) do
+            {:ok, conn} -> {:ok, conn}
+            {:error, _, _} -> :closed
+          end
+
+        _ ->
+          {:ok, conn}
```

Comment:
Probably best to not change the code style to keep unrelated changes out of the pull request.